vignettes/orderly2.Rmd

```diff
@@ -0,0 +1,81 @@
+* a system of storage so that results of running analysis can be shared among a group
+* a system to control how integration of these pieces of analysis takes place
+
+We refer to a transportable unit of analysis as a "packet". This conceptually is a directory of files created by running some code, and is our atomic unit of work from the point of view of `orderly`.  Each packet has an underlying source form, which anyone *can* run. However, most of the time people will use pre-run packets and 
+
+# Reproducible analyses
+
+Any degree of collaboration in the style above requires reproducibility, but there are several aspects of this.
+
+With the system we describe here, even though everyone *can* typically run any step of an analysis, they typically *don't*. This differs from workflow tools, which users may be familiar with.
+
+## Difference from a workflow system
+
+Workflow systems have been hugely influential in scientific computing, from people co-opting build systems like [make](https://en.wikipedia.org/wiki/Make_(software)) through to sophisticated systems designed for parallel running of large and complex workflows such as [nextflow](https://www.nextflow.io/). The general approach is to define interdependencies among parts of an analysis, forming a [_graph_](https://en.wikipedia.org/wiki/Directed_acyclic_graph) over parts of an analysis and track inputs and outputs through the workflow. 
+
+This model of computation has lots of good points:
+
+* it defines an interface over an analysis and allows (and encourages) breaking a monolithic analysis into component pieces which can be reasoned about
+* it allows high-level parallelism by making obvious parts of the workflow that can be run concurrently
+* by tracking the way data flows through an analysis, it allows the minimum amount of recalculation to be done on change, with only downstream parts triggered
+* with a shared workspace or online runner, allows a degree of collaboration so long as everyone is happy to be working with a constantly changing set of code and analysis artefacts
+
+We have designed `orderly2` for working patterns that do not suit the above. Some motivating reasons include:
+
+* Some nodes in the computational graph are very expensive to compute or require exotic hardware.
+* Workflows where the upstream data never settle, but we need to know which version of the data ends up used in a particular analysis
+* Workflows where upstream analyses are used in many downstream analyses, and where the upstream developer may not know much about the downstream use
+* Nondeterministic analyses, e.g., those involving stochastic simulations, where rerunning a node is not expected to return the exact same numerical results, and so we can't rely on different users recovering the same results on different occasions.
```

Comment:
is it worth noting that it is possible to set the seed ie. completely recover the results?